### PR TITLE
Netty Pub/Sub Clients & Ack messages & broke-common module

### PR DESCRIPTION
**Pull request by allenlee820202**
Netty Pub/Sub Clients
1. Broke block client now uses netty.
2. Send messages using Netty channel.
3. Handle ack messags using AckHandler, a channel inbound handler
adapter

Ack messages
1. Now server acks CONNECT, PUBLISH, SUBSCRIBE messages. The ack
messages are named CONNACK, PUBACK, SUBACK respectively.

broke-common module
1. Common class that shares between different modules can place in this
module
2. The MessageParseHandler is now moved to this module. Because both
server and client shares such channel handler.

@tz70s please help take a look, thanks!

### CODE REVIEW

File: broke-protocol/src/main/java/org/dsngroup/broke/protocol/Message.java [hunk 1]
```diff
@@ -36,9 +36,9 @@
     // Message fields
     // TODO: may let this message to release the burden of construction
     // and destruction.
-    protected Method method;
-    protected MessageHeader header;
-    protected Object payload;
+    protected final Method method;
+    protected final MessageHeader header;
+    protected final Object payload;
```
Comment (tz70s) [inline broke-protocol/src/main/java/org/dsngroup/broke/protocol/Message.java]: I'm wondering if further we can eliminate the construction and destruction burden from these fields. Maybe `final` comes with the lower flexibility.

File: broke-client/src/main/java/org/dsngroup/broke/client/BlockClient.java [hunk 2]
```diff
@@ -34,37 +38,70 @@
 
     private int targetBrokerPort;
 
-    private Socket clientSocket;
+    private Channel myChannel;
+
```
Comment (tz70s) [inline broke-client/src/main/java/org/dsngroup/broke/client/BlockClient.java]: Replace `myChannel` into a more verbose or clear name.

**Comment by tz70s** [comment]
Nice work! Check out some requested modification, thx!

File: broke-protocol/src/main/java/org/dsngroup/broke/protocol/Message.java (hunk 1 above)
Comment (allenlee820202) [reply broke-protocol/src/main/java/org/dsngroup/broke/protocol/Message.java]: OK, I'll remove the ```final``` keyword.

**Comment by allenlee820202** [comment]
The modification is done. Please check again, thanks!

**Comment by tz70s** [comment]
Merge, thanks!

**Pull request merged**